### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by faizi2500**
### Following tasks achieved in this milestone:

- [x] Add RSpec as a gem to your project.
- [x] Create a Request spec file for all your controllers
- [x] Make sure that for each action you check:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    end
+
+    it 'renders correct template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'Posts of user'
+    end
+  end
```
Comment (Emmyn5600) [inline spec/requests/posts_spec.rb]: - [ ] Test that it does not render a different template
- [ ] Let us change the .to include in this way `expect(response.body).to include("Posts of user")`

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    it 'renders correct template' do
+      expect(response).to render_template('show')
+    end
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'Posts 4 by user'
+    end
+  end
+end
```
Comment (Emmyn5600) [inline spec/requests/posts_spec.rb]: - [ ] Check that it does not render a different template
- [ ] Modify `expect(response.body).to include 'Posts 4 by user'` with `expect(response.body).to include ("Posts 4 by user")`

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      expect(response).to render_template('index')
+    end
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'All Users'
+    end
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'Here is a list of all users'
+    end
```
Comment (Emmyn5600) [inline spec/requests/users_spec.rb]: - [ ] Change  `expect(response.body).to include 'Here is a list of all users'` to `expect(response.body).to include ('Here is a list of all users')`

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'Users ID 745'
+    end
+
+    it 'check for placeholder text' do
+      expect(response.body).to include 'This user details'
+    end
+  end
```
Comment (Emmyn5600) [inline spec/requests/users_spec.rb]: - [ ]  Change `expect(response.body).to include 'This user details'`  to ` expect(response.body).to include ('This user details')`

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @faizi2500 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.


 ## Required Changes ♻️
 
_Check the comments under the review._

## Optional suggestions

- N/A 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (faizi2500) [reply spec/requests/posts_spec.rb]: Hi @Emmyn5600, thanks for mentioning the issues. I have fixed these.

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (faizi2500) [reply spec/requests/posts_spec.rb]: Hi @Emmyn5600, thanks for mentioning the issues. I have fixed these.

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (henatan99) [reply spec/requests/posts_spec.rb]: These issues have been addressed well. Good Job! :clap: :clap:

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (henatan99) [reply spec/requests/posts_spec.rb]: These issues have been addressed well. Good Job! :+1: :+1:

File: spec/requests/users_spec.rb (hunk 3 above)
Comment (henatan99) [reply spec/requests/users_spec.rb]: This issue has been addressed well. Nice work! :+1: :+1:

File: spec/requests/users_spec.rb (hunk 4 above)
Comment (henatan99) [reply spec/requests/users_spec.rb]: This issue has been addressed well. Nice work! :clap: :clap:

**Review by henatan99 — APPROVED** [review]
Hi @faizi2500 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**